tests/ceph_ansible/test_ansible_upgrade.py

```diff
@@ -179,9 +179,15 @@ def get_container_counts(ceph_cluster):
         distro_ver = distro_info['VERSION_ID']
         if distro_ver.startswith('8'):
             out, rc = node.exec_command(sudo=True, cmd='podman ps | grep $(hostname) | wc -l')
+            # In ceph 4.2 onwards ceph-crash as new container got added
+            # so decreasing that count to pass compare_container_counts function
+            crash, rc = node.exec_command(sudo=True, cmd='podman ps |grep ceph-crash| wc -l')
```

Comment:
as per your comment. the extra container is from 4.2 on wards. I think , you need to change it compare ceph version. what do you suppose will happen if 4.2 is run on a RHEL 7.x ?

Reply:
@rakeshgm Initially I did the changes in compare_container_count function But this solution won't work If anyone wants to upgrade from 3.x to 4.1 also for every new release this code has to get changed with version
So to avoid this Just get the crash containers count inside the node itself
also as per my PR commit it should handle gracefully for rhel 7 and rhel 8 since i have used docker and podman commands respectively
```
-def compare_container_counts(pre_upgrade_counts, post_upgrade_counts, prev_install_version):
+def compare_container_counts(pre_upgrade_counts, post_upgrade_counts, prev_install_version, post_upgrade_versions):
     """
     Compare pre-upgrade and post-upgrade container counts.
 
@@ -202,14 +203,22 @@ def compare_container_counts(pre_upgrade_counts, post_upgrade_counts, prev_insta
     """
     log.info("Pre upgrade container counts: {}".format(pre_upgrade_counts))
     log.info("Post upgrade container counts: {}".format(post_upgrade_counts))
-
+    # In ceph 4.2 onwards ceph-crash as new container got added
+    # in these daemons mon, mgr, mds, rgw, osd, fuse, rbd-mirror
+    crash_collocated_daemons = ['mon', 'mgr', 'mds', 'rgw', 'osd', 'fuse', 'rbd-mirror']
     for node, count in post_upgrade_counts.items():
         if prev_install_version.startswith('2'):
             # subtract 1 since mgr containers are now collocated on mons
             if '-mon' in node:
                 count -= 1
+
+        # adding that ceph-crash container count in pre upgrade count list
+        if post_upgrade_versions.startswith('4'):
+            if any(daemon in node for daemon in crash_collocated_daemons):
+                pre_upgrade_counts[node] += 1
         if pre_upgrade_counts[node] != count:
-            log.warning("Mismatched container count post upgrade")
+            log.error("Mismatched container count post upgrade")
+            return 1
     return 0
```

But It got failed  for rhel 7.x with
 `2020-10-22 13:17:14,581 - ceph.ceph - INFO - 
2020-10-22 13:17:14,582 - test_ansible_upgrade - ERROR - Failed during upgrade (rc = 2)
2020-10-22 13:17:14,582 - __main__ - INFO - Test <module 'test_ansible_upgrade' from '/home/sangadi/cephci/cephci/tests/ceph_ansible/test_ansible_upgrade.py'> failed`
But not sure where it went wrong 
Logs : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1603369242279/result.html
@rakeshgm @sunilkumarn417 Please let me know your thougts on this from where this error is coming?

Reply:
@sunilangadi2 I would suggest to go with further classification(with daemon/roles) in node as below to overcome this issue.

Assuming this scenario without any scaling tests,
before upgrade( node-1 : mon_count:1, mgr_count:1, osd_count:2)
after upgrade(node-1: mon_count:1, mgr_count:1, osd_count:2), 

and any new containers are created post-upgrade like ceph-crash, consider it for with release version post-upgrade.

Reply:
Done